### PR TITLE
Tests to verify exit codes and terminal setup

### PR DESCRIPTION
**Pull request by chavafg**
This PR contains two commits that add integration tests to verify exit codes and terminal setup:

integration tests: Add tests to verify that 
    1. a pseudo tty is properly setup when it is requested
    2. TERM environment variable is set when a tty is requested
    3. TERM environment variable is not set when a tty is not requested

integration tests: Add test to verify exit code from a container
Add tests to verify that a container returns the correct exit code of the executed process.

### CODE REVIEW

**Comment by chavafg** [comment]
qa-failed

**Comment by dlespiau** [comment]
Considering your comment in #54, it means those tests are all passing? I'd suggest we skip the ones that are known to fail, we shouldn't add failing tests before they are fixed.

**Comment by sboeuf** [comment]
@chavafg at least you should tag this PR as Work In Progress until you have those tests fixed. Or you remove them as @dlespiau suggested.

**Comment by chavafg** [comment]
done :)

**Comment by chavafg** [comment]
After research, found that these tests are working, 
```
1..2 
ok 1 Exit Code from container process when running non-interactive 
ok 2 Exit Code from container process when running interactive 
	
1..3 
ok 1 TERM env variable is set when allocating a tty 
ok 2 TERM env variable is not set when not allocating a tty 
ok 3 Check that pseudo tty is setup properly when allocating a tty
```
but there is an issue which breaks the terminal #527 

can we merge this tests?

File: tests/integration/docker/exit-code.bats [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+exit_status=55
+
+setup() {
+	source $SRC/test-common.bash
+	clean_docker_ps
+	runtime_docker
+}
+
+@test "Exit Code from container process when running non-interactive" {
+	run $DOCKER_EXE run ubuntu /usr/bin/perl -e "exit $exit_status"
```
Comment (devimc) [inline tests/integration/docker/exit-code.bats]: I think that you could use the perl container

**Comment by sameo** [comment]
LGTM

**Comment by devimc** [comment]
lgtm

**Pull request merged**